### PR TITLE
Add activation_function and diffusive options to ConvolutionalNetworkConfig

### PR DESCRIPTION
**Pull request by mcgibbon**
This PR adds configuration options to ConvolutionalNetworkConfig necessary to make it perform learned diffusion.

Refactored public API:
- Added options `diffusive` and `activation_function` to `ConvolutionalNetworkConfig`

- [x] Tests added

(Delete this for the commit message)
You are encouraged to check the PR against the [Code Review Checklist](https://paper.dropbox.com/doc/Code-Review-Checklist--A4lKrs~xg7w5Gsb39N6JLNQoAg-IlsYffZgTwyKEylty7NhY).

### CODE REVIEW

File: external/fv3fit/fv3fit/keras/_models/shared/convolutional_network.py [hunk 1]
```diff
@@ -26,6 +26,24 @@ def __call__(self, w: tf.Tensor):
         return tf.scalar_mul(0.5, w + tf.transpose(w, perm=(1, 0, 2, 3)))
 
 
+class Diffusive(tf.keras.constraints.Constraint):
+    def __call__(self, w: tf.Tensor):
+        w = tf.maximum(w, 0.0)
+        total = tf.reduce_sum(w)
+        return tf.scalar_mul(1.0 / total, w)
```
[the remark this answers is not in the document]
Comment (mcgibbon) [reply external/fv3fit/fv3fit/keras/_models/shared/convolutional_network.py:33]: I can set `use_bias=False` when `diffusive=True`, and add the RadialConstraint.

One thing you'll need to keep in mind that I tried to document is that if the normalization is different for your inputs and outputs, then this still won't be conservative. Granted, that non-conservation may be desirable, given it's rescaling the input domain to look more like the output domain.

File: external/fv3fit/tests/training/test_train.py [hunk 2]
```diff
@@ -203,6 +206,37 @@ def test_default_convolutional_model_is_transpose_invariant(regtest):
     xr.testing.assert_allclose(output_from_transpose, transpose_output, atol=1e-5)
 
 
+def test_diffusive_convolutional_model_gives_bounded_output():
+    """
```
[the remark this answers is not in the document]
Comment (mcgibbon) [reply external/fv3fit/tests/training/test_train.py:210]: The means will not be the same because the mean of the input (when fit to the normalizer) is not the mean of the output. But given the input, the mean of the output *should* be almost exactly equal to the mean of the target output. Unfortunately this is not very worthwhile to test since this will be true of almost any mildly-trained model.

**Review by brianhenn — APPROVED** [review]
Thanks @mcgibbon
